plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveMetadata.java

```diff
@@ -1578,7 +1578,7 @@ public ConnectorTableHandle beginUpdate(ConnectorSession session, ConnectorTable
         Table table = metastore.getTable(identity, tableName.getSchemaName(), tableName.getTableName())
                 .orElseThrow(() -> new TableNotFoundException(tableName));
 
-        if (!isTransactionalTable(table.getParameters())) {
+        if (!isFullAcidTable(table.getParameters())) {
```

Comment:
It would be great to have a test (& for DELETE too)

Reply:
This is a bug that should be fixed no matter what -- please create a separate PR so we can merge it independently of the feature flag discussion

Reply:
https://github.com/trinodb/trino/pull/7607